### PR TITLE
Start migrating scanner plugins to separate projects

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -106,61 +53,32 @@ project(":clients:osv").name = "osv-client"
...
 project(":utils:scripting").name = "scripting-utils"
 project(":utils:spdx").name = "spdx-utils"
 project(":utils:test").name = "test-utils"
 
+file("plugins").walk().maxDepth(3).filter {
+    it.isFile && it.name == "build.gradle.kts"
+}.mapTo(mutableListOf()) {
+    it.parentFile.toRelativeString(rootDir).replace(File.separatorChar, ':')
+}.forEach {
+    include(":$it")
```
Comment (fviernau) [inline settings.gradle.kts:67]: It wasn't super obvious to me what these two `it`s in line 65 and 67 refer to.
How about fixing this by using some variable, like e.g:

```
val moduleNames = file("plugins").walk().maxDepth(3).filter {
    it.isFile && it.name == "build.gradle.kts"
}.mapTo(mutableListOf()) {
    ":${it.parentFile.toRelativeString(rootDir).replace(File.separatorChar, ':')}" 
}

moduleNames.forEach { moduleName ->
     include(moduleName)
     ...
}
```

File: settings.gradle.kts [hunk 2]
```diff
@@ -106,61 +53,32 @@ project(":clients:osv").name = "osv-client"
...
+    val parts = it.split(':')
+    if (parts.size == 3 && !parts[2].startsWith("gradle-") && !parts[2].endsWith("-template")) {
+        val singularName = parts[1].removeSuffix("s")
+
+        project(":$it").name = if (parts[2] == "api") {
+            "$singularName-api"
+        } else {
+            "${parts[2]}-$singularName"
+        }
+    }
```
Comment (fviernau) [inline settings.gradle.kts:79]: To me it's hard to grasp what

1. The condition in line 71 is about. Would it make sense to replace it with a hard coded ignore list for
     readability?
2. The reason why the "s" is stripped from the name in line 72

In general, would it make sense to add some code comment before line 62 (stating the purpose of the below code block) and to the logic in the function, to explain a bit the what and why / idea?

File: plugins/commands/scanner/build.gradle.kts [hunk 3]
```diff
@@ -25,6 +25,8 @@ plugins {
 dependencies {
     api(project(":plugins:commands:command-api"))
 
+    implementation(platform(project(":plugins:scanners")))
```
Comment (fviernau) [inline plugins/commands/scanner/build.gradle.kts:28]: commit-msg: "This changes requires" -> "This change requires"?

**Comment by fviernau** [comment]
LGTM now ok from my side. Maybe we can have one more reviewer? @oss-review-toolkit/core-devs ?

File: settings.gradle.kts [hunk 4]
```diff
@@ -106,61 +53,36 @@ project(":clients:osv").name = "osv-client"
...
+    val parts = projectPath.split(':')
+    if (parts.size == 3 && parts[2] !in accompanyingProjects) {
+        // Convert the plural name for the type of plugin to singular.
+        val singularTypeName = parts[1].removeSuffix("s")
+
+        project(":$projectPath").name = if (parts[2] == "api") {
+            "$singularTypeName-api"
+        } else {
+            "${parts[2]}-$singularTypeName"
+        }
```
Comment (mnonnenmacher) [inline settings.gradle.kts:82]: I think a when expression would look nicer here:
```kotlin
        project(":$projectPath").name = when(parts[2]) {
            "api" -> "$singularTypeName-api"
            else -> "${parts[2]}-$singularTypeName"
        }
```

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**